ComposeSamples/CompanionPane/src/main/java/com/microsoft/device/display/samples/companionpane/HomePage.kt

```diff
@@ -93,97 +102,132 @@ fun SetupUI(windowInfoRep: WindowInfoRepository) {
-    }
+    TwoPaneLayout(
+        pane1 = { Pane1(screenState) },
+        pane2 = { Pane2(screenState) },
+    )
 }
 
 @Composable
-fun DualPortraitLayout() {
-    Row(
-        modifier = Modifier
-            .fillMaxSize()
-            .verticalScroll(rememberScrollState())
-            .padding(top = 10.dp, bottom = 10.dp),
-        horizontalArrangement = Arrangement.spacedBy(50.dp),
-    ) {
-        Column(
-            modifier = Modifier
-                .fillMaxHeight()
-                .weight(1f),
-            verticalArrangement = Arrangement.SpaceEvenly
-        ) {
-            ImagePanel(
-                Modifier.padding(top = 20.dp, bottom = 20.dp, start = 20.dp, end = 20.dp)
+fun ShowWithTopBar(content: @Composable () -> Unit, title: String? = null) {
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = {
+                    BasicText(
+                        text = title ?: "",
+                        style = TextStyle(
+                            fontSize = 18.sp,
+                            fontWeight = FontWeight.Bold,
+                            color = MaterialTheme.colors.onPrimary
+                        )
+                    )
+                },
+                backgroundColor = MaterialTheme.colors.primaryVariant
             )
-            EffectPanel()
-        }
-        Column(
-            modifier = Modifier
-                .fillMaxHeight()
-                .weight(1f),
-            horizontalAlignment = Alignment.CenterHorizontally
-        ) {
-            Column(
-                modifier = Modifier.padding(top = 20.dp),
-                verticalArrangement = Arrangement.spacedBy(35.dp)
-            ) {
-                MagicWandPanel(modifier = Modifier.width(longSlideWidth))
-                DefinitionPanel(modifier = Modifier.width(longSlideWidth))
-                VignettePanel(modifier = Modifier.width(longSlideWidth))
-                BrightnessPanel(modifier = Modifier.width(longSlideWidth))
+        },
+        content = { content() }
+    )
+}
+
+@Composable
+fun Pane1(screenState: ScreenState) {
+    ShowWithTopBar(
+        content = {
+            when (screenState) {
+                ScreenState.SinglePortrait -> PortraitLayout()
+                ScreenState.SingleLandscape -> LandscapeLayout()
+                ScreenState.DualPortrait -> DualPortraitPane1()
+                ScreenState.DualLandscape -> DualLandscapePane1()
             }
-            Spacer(Modifier.height(80.dp))
-            ShortFilterControl()
+        },
+        title = stringResource(R.string.app_name)
+    )
+}
+
+@Composable
+fun Pane2(screenState: ScreenState) {
+    when (screenState) {
+        ScreenState.DualPortrait -> {
+            ShowWithTopBar(
+                content = { DualPortraitPane2() }
+            )
         }
+        ScreenState.DualLandscape -> DualLandscapePane2(Modifier.background(color = MaterialTheme.colors.background))
```

Comment:
I think you can just set `Modifier.background(color = MaterialTheme.colors.background` to `Column` inside `DualLandscapePane2`. No need to abstract the `modifier` as the parameter.